core/src/main/java/org/apache/iceberg/RemoveSnapshots.java

```diff
@@ -365,17 +365,13 @@ ExpireSnapshots withIncrementalCleanup(boolean useIncrementalCleanup) {
   private void cleanExpiredSnapshots() {
     TableMetadata current = ops.refresh();
 
-    if (specifiedSnapshotId) {
-      if (incrementalCleanup != null && incrementalCleanup) {
-        throw new UnsupportedOperationException(
-            "Cannot clean files incrementally when snapshot IDs are specified");
-      }
-
-      incrementalCleanup = false;
-    }
-
-    if (incrementalCleanup == null) {
-      incrementalCleanup = current.refs().size() == 1;
+    if (Boolean.TRUE.equals(incrementalCleanup)) {
+      validateCleanupCanBeIncremental(current);
+    } else {
+      incrementalCleanup =
```

Comment:
This doesn't properly handle the case where incremental cleanup was explicitly set to false